theme.js

```diff
@@ -0,0 +1,105 @@
+/* jshint node: true */
+'use strict';
+
+var Addon = require('ember-cli/lib/models/addon');
+
+var path = require('path');
+var csso = require('broccoli-csso');
+
+var jsPreprocessor = require('./lib/js-preprocessor');
+var templatePreprocessor = require('./lib/template-preprocessor');
+var scssPreprocessor = require('./lib/scss-preprocessor');
+
+var TransformComponentClasses = require('./lib/transform-component-classes');
+var TransformUITableComponents = require('./lib/plugins/transform-ui-table-components');
+
+var UITheme = Addon.extend({
+  setupPreprocessorRegistry: function(type, registry) {
+    this.setupJsPreprocessing(registry, type);
+
+    if (type === 'self') {
+      this.setupTemplatePreprocessing(registry);
+      this.setupCssPreprocessing(registry);
+      this.setupHtmlTransform(registry);
+    }
+
+    registry.add('htmlbars-ast-plugin', {
+      name: 'transform-ui-table-components',
+      plugin: TransformUITableComponents
+    });
+  },
+
+  setupJsPreprocessing: function(registry, type) {
+    registry.add('js', {
+      name: 'generate-components-for-scss',
+      ext: 'scss',
+      _addon: this,
+      toTree: function(tree) {
+        var destDir = tree.destDir || (tree.inputTree && tree.inputTree.destDir);
+        var isTests = /tests$/.test(destDir);
+        var isSelf = type === 'self';
+        var isParent = type === 'parent';
+
+        var options = {
+          registryType: type,
+          moduleName: this._addon.name,
+          prefix: isSelf ? ('modules/' + this._addon.name) : registry.app.name,
+          baseDir: path.join(this._addon.root, 'addon'),
```

Comment:
@ianstarz `this._addon.addonRoot()` was previously being used here and below instead of `this._addon.root`. I'm not sure if there was a specific reason but the `addonRoot()` function was relying on `__dir`, which wasn't correct when this was being used inside the `ui-base-theme` addon, because it was still using this addon's directory.

As far as I could tell, `this.root` was giving the correct path and seems to work.